internal/framework/schemas.go

```diff
@@ -97,6 +101,28 @@ func IsImmutableAttribute(oas *openapi3.Schema) bool {
 	return IsExtensionSet(oas, "x-immutable")
 }
 
+func GetStringValidators(oas *openapi3.Schema) []validator.String {
+	var validators []validator.String
+
+	format := oas.Format
+	if format == "date-time" {
+		timestampPattern := "^([0-9]+)-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])[Tt]([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9]|60)(([Zz])|([\\+|\\-]([01][0-9]|2[0-3]):[0-5][0-9]))$"
+		validators = append(validators, stringvalidator.RegexMatches(regexp.MustCompile(timestampPattern), "must be a RFC 3339 date-time without fractional seconds"))
+	}
+
+	pattern := oas.Pattern
+	if pattern != "" {
+		if !strings.HasPrefix(pattern, "^") {
+			pattern = "^" + pattern
+		}
+		if !strings.HasSuffix(pattern, "$") {
+			pattern = pattern + "$"
+		}
```

Comment:
Technically we should already have these in the pattern. I will add a CP bug about it.